contracts/events/TopicEvent.sol

```diff
@@ -118,14 +118,12 @@ contract TopicEvent is ITopicEvent, Ownable {
         version = _version;
         owner = _owner;
         name = _name;
-        resultNames = _resultNames;
 
+        resultNames[0] = "Invalid";
+        numOfResults++;
         for (uint i = 0; i < _resultNames.length; i++) {
             if (!_resultNames[i].isEmpty()) {
-                balances[i] = ResultBalance({
-                    totalBets: 0,
-                    totalVotes: 0
-                    });
+                resultNames[i + 1] = _resultNames[i];
```

Comment:
This could leave empty item in resultNames array. If value is empty you skip it, then the corresponding slot in resultNames is empty as well, so the check doesn't really do anything. Alternatively, we can use another index j as resultNames index, so when _resultName[i] is empty, we don't increment j, thus leave no empty item in the result array.

It's a design question.

Reply:
I think misunderstanding `break` in Solidity. It actually breaks out of the entire loop so when the loop finds an empty result name, it will stop the loop there.

Also, either UI or API side should be handling the case where there is an empty slot in between non-empty result names and remove the empty slot.